File: Java/0541-reverse-string-ii.java
class Solution {
    public String reverseStr(String s, int k) {
        int k_2 = k << 1;
        char[] arr = s.toCharArray();
        for (int i = 0; i < s.length(); i += k_2) {
            int l = i, r = Math.min(l + k - 1, s.length() - 1);
            while (l < r) {
                arr[l] ^= arr[r];
                arr[r] ^= arr[l];
                arr[l] ^= arr[r];
                l++;
                r--;
            }
        }

        return new String(arr);
    }
}

File: Python/0541-reverse-string-ii.py
class Solution:
    def reverseStr(self, s: str, k: int) -> str:
        k_2 = k << 1
        arr = list(s)
        for i in range(0, len(s), k_2) :
            l, r = i, min(i + k - 1, len(s) - 1)
            while l < r :
                arr[l], arr[r] = arr[r], arr[l]
                l += 1
                r -= 1
        return ''.join(arr)
                    